src/core/thread/network_data.cpp

```diff
@@ -686,18 +681,6 @@ const PrefixTlv *NetworkData::FindPrefix(const uint8_t *aPrefix,
     return prefixTlv;
 }
 
-int8_t NetworkData::PrefixMatch(const uint8_t *a, const uint8_t *b, uint8_t aLength)
-{
-    uint8_t matchedLength;
-
-    // Note that he `Ip6::Address::PrefixMatch` expects the prefix
-    // length to be in bytes unit.
-
-    matchedLength = Ip6::Address::PrefixMatch(a, b, BitVectorBytes(aLength));
-
-    return (matchedLength >= aLength) ? static_cast<int8_t>(matchedLength) : -1;
-}
```

Comment:
Thanks @jwhui  for reviewing this. I'd like to ask for your help checking something.

In this commit/PR, I removed the `NetworkData` version of ` PrefixMatch` and used the new helper methods where it was being used before.

The `NetworkData::PrefixMatch()` calculates the `matchedLength` and if larger than given `aLength` it would return the `matchedLength` which can potentially be larger than `aLength`.

There is one place that removing this may change the existing behavior. 

- In `LeaderBase::ExternalRouteLookup()` we go through all the `PrefixTlv` from network data and try to match them with a given `aDestination` address and keep track of the best match so far.
- I changed the code in this PR to use `Address::MatchPrefix()` and then use the prefix length itself as the match length.
- This would be different from `NetworkData::PrefixMatch` which can potentially use a longer match length than  prefix length. 
- I think the new model is better and correct but just want to double-check with you as well.  

Thanks. :)

Reply:
I agree. Thanks for catching this!